File: examples/example-py/xk6-output-plugin-example.py
#!/bin/env python

import datetime
import logging

from xk6_output_plugin_py.output import serve, Output, Info, MetricType, ValueType


class Example(Output):
    def Init(self, params):
        logging.info("init")

        return Info(description="example-py plugin")

    def Start(self):
        logging.info("start")

    def Stop(self):
        logging.info("stop")

    def AddMetrics(self, metrics):
        logging.info("metrics")
        for metric in metrics:
            logging.info(
                metric.name,
                extra={
                    "metric.type": MetricType.Name(metric.type),
                    "metric.contains": ValueType.Name(metric.contains),
                },
            )

    def AddSamples(self, samples):
        logging.info("samples")
        for sample in samples:
            t = datetime.datetime.fromtimestamp(
                sample.time / 1000.0, tz=datetime.timezone.utc
            )

            logging.info(
                sample.metric,
                extra={"sample.time": t, "sample.value": sample.value},
            )


if __name__ == "__main__":
    serve(Example())

File: examples/example-js/xk6-output-plugin-example.js
#!/usr/bin/env node

import { serve } from 'xk6-output-plugin-js'

class Example {
  init (params) {
    console.info('init')
    return { description: 'example-js plugin' }
  }

  start () {
    console.info('start')
  }

  stop () {
    console.info('stop')
  }

  addMetrics (metrics) {
    console.info('metrics')
    metrics.forEach(metric => {
      console.info(
        { 'metric.type': metric.type, 'metric.contains': metric.contains },
        metric.name
      )
    })
  }

  addSamples (samples) {
    console.info('samples')
    samples.forEach(sample => {
      console.info(
        { 'sample.time': new Date(sample.time).toISOString(), 'sample.value': sample.value },
        sample.metric
      )
    })
  }
}

serve(new Example())
